lib/funneler.rb

```diff
@@ -30,14 +30,17 @@ def from_token(token:)
       raise InvalidTokenError, "Invalid token '#{token}': #{e.message}"
     end
 
-    def build(route_generator:, params: {}, expires_in_days: nil)
+    def build(route_generator:, params: {}, expires_in_days: nil, context: {})
```

Comment:
I think `meta` is a better name than `context` here, `context` maybe should be the key something inside the meta hash...Thinking of the way we could/will use this in the future (tracking, auth, trigger actions, set flash messages etc), it should be a bit more abstract.

I agree that the current app-side implementation of `context` is a little clunky, ie:

```ruby
-    funnel = Funneler.from_routes(routes: routes)
+    funnel = Funneler.from_routes(routes: routes,
+                                  context: { funnel_name: Funnels::PendingInvitesStue.name })

```

Default contextual information (like the funnel name) should be added to `Funnels::Base` and perhaps the more information can be captured here in the constructor, eg, `context.reverse_merge!(caller: caller.last)`